compiler/rustc_infer/src/infer/region_constraints/mod.rs

```diff
@@ -617,8 +623,25 @@ impl<'tcx> RegionConstraintCollector<'_, 'tcx> {
         }
     }
 
-    pub fn opportunistic_resolve_var(&mut self, rid: RegionVid) -> ty::RegionVid {
-        self.unification_table().probe_value(rid).min_vid
+    pub fn opportunistic_resolve_var(&mut self, rid: ty::RegionVid) -> ty::RegionVid {
```

Comment:
We should comment on the distinction i behavior between these two functions!

Reply:
Added